### DISCUSSION TITLE
Add mock response without writing example in openapi spec

### DISCUSSION THREAD

**Discussion by maestrus** [microcks/microcks#1401] [Community]
Hi, I would like to know if it should be possible to add mock response once imported openapi without providing example.
1) Create openapi without example
2) import openapi into microcks
3) add mockresponse

Thanks

**Comment by lbroudoux** [comment]
Hey! You can do that using our AICopilot / OpenAI or other LLM integration. You can read a post on this here: https://medium.com/itnext/boost-your-api-mocking-workflow-with-ollama-and-microcks-38e25fe78450

**Comment by maestrus** [comment]
Hi @lbroudoux so this is the only way to add mock response? I don't want to expose in my yaml example and i can not  use LLM integration at the moment  and i want my QA team feels free to add all mock they needs. Should be possible to use an api to add mock response to an endpoint? or maybe in next release , add a ui button to add custom mock? thank you

**Reply by lbroudoux** [reply]
No, this is not the only way, it's just the only way to do it interactively via the UI.

Microck's principle is to rely on existing artifacts that can be put under source control to create a reproducible and trackable source of truth. We have what we call the [multi-artifacts support](https://microcks.io/documentation/explanations/multi-artifacts/) so that it's not mandatory to put your examples into your OpenAPI file.

You can consider having your OpenAPI file defines the structure (operations, parameters, schemas, etc...) and then add other artifacts to bring in examples. You can use:
* Postman Collection, (see https://microcks.io/documentation/references/artifacts/postman-conventions/)
* HTTP Archive - from live traffic recording for example, (see https://microcks.io/documentation/references/artifacts/har-conventions/)
* Our own lightweight APIExample format (see https://microcks.io/documentation/references/examples/)

**Reply by maestrus** [reply]
ok thank you , i will dig into.
Giorgio
